### ISSUE TITLE
SlnFile or similar API should be public

### ISSUE THREAD

**Issue by hknielsen** [dotnet/sdk#28516]
### Is your feature request related to a problem? Please describe.
Im looking for a Sln file generator, and maintance library, to create sln, and add/remove projects + configurations.

As I know that `dotnet new sln -o ...` was doing most of this, I was hoping there were a public API somewhere to share the logic, for future changes and handling.

SlnFile that is used by `dotnet new sln...` and ect. should be extracted to a nuget package.
I see two copies of the class just in the dotnet github org
https://github.com/search?q=org%3Adotnet+%22class+SlnFile%22&type=code

### Describe the solution you'd like
Extract `SlnFile`, or something similar to a nuget package that other systems also uses so we can share the logic.
Microsoft.Build would maybe be a suitable place.

**Comment by baronfel** [comment]
Solution file parsing is also available in the existing MSBuild packages in the `Microsoft.Build.Construction` namespace I believe. I would lean on that for your work, as that's actually supported. I would expect any other code you find to be domain-specific to their respective tools.

**Comment by hknielsen** [comment]
@baronfel Yes `SolutionFile.Parse` can do Sln file parsing/reading. But adding project's or configurations to the solution, and writing down the solution again is lacking, compared to the `SlnFile` class.

I would like to, from code:
- Create Sln
- Add/Remove projects
- Update Configurations

Currently I dont see I can do that with `Microsoft.Build` or other public API? The closest I see is the `SlnFile` thats used by the cli

**Comment by marcpopMSFT** [comment]
Realistically, the solution file format is actually owned by Visual Studio. We'll share the feedback with them as we'd like if they made it open source and provided apis that we could use as well but for now, they don't have plans to afaik.
